### ISSUE TITLE
Support calls to enum classes

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7027]
When an enum class is created, the `EnumType` metaclass replaces its original constructor with a new implementation that performs a value-based lookup. The following should therefore type check without errors.

```python
from enum import Enum

class Example(Enum):
    A = (1, 2)

    def __init__(self, value, other) -> None:
        self._value_ = value
        self.other = other

Example(1)  # This should not generate an error
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.348, which was just published. It will also be included in a future release of pylance.
